README.md

```diff
@@ -0,0 +1,172 @@
+
+# Fuel Telemetry
+
+`fuel-telemetry` is a `tracing` `Layer` used to implement telemetry within our
+apps and libraries.
+
+Steps to get `fuel-telemetry` going:
+
+- Create a `TelemetryLayer` `tracing` `Layer`
+- Add the `TelemetryLayer` to a `tracing` `Subscriber`
+- Enable the `telemetry` field on `Span`s you want sent to InfluxDB
+- Use the regular `tracing` macros (e.g `info!()`, `warn!()` etc) to record `Events`
+- Start a `FileWatcher` so that telemetry files will be sent to InfluxDB
+- Start a `SystemInfoWatcher` so that system info will be recorded
+
+Most of these steps are hidden away behind convenience functions within `TelemetryLayer`.
+
+## Using `fuel-telemetry`
+
+### Within Applications with No Existing Tracing (Enabled by Default)
+
+If you're writing an app where:
+
+- `fuel-telemetry` will be your only `tracing` `Layer` and `Subscriber`
+- and **you want all** `tracing` `Event`s to be sent to InfluxDB by default
+
+then the quickest way to get going is:
+
+```rust
+use fuel_telemetry::prelude::*;
+
+fn main() {
+    telemetry_init().unwrap();
+
+    info!("This event will be sent to InfluxDB");
+}
+```
+
+### Within Applications with No Existing Tracing (Disabled by Default)
+
+If you're writing an app where:
+- `fuel-telemetry` will be your only `tracing` `Layer` and `Subscriber`
+- and **you do NOT want all** `tracing` `Event`s to be sent to InfluxDB by default
+
+then the quickest way to get going is:
+
+```rust
+use fuel_telemetry::prelude::*;
+
+fn main() {
+    let _guard = TelemetryLayer::new_global_default_with_watchers()?;
+    info!("This event will not be sent in InfluxDB");
+
+    // ... <code containing tracing events with telemetry disabled>
+    // ...
+    // ...
+
+    // Time to enable telemetry from now on by setting telemetry=true on a `Span`
+    let enabled_span = span!(Level::INFO, "root", telemetry=true);
+    let _span_guard = enabled_span.enter();
```

Comment:
can we reduce this to 1 line?

Reply:
https://docs.rs/tracing/latest/tracing/struct.Span.html#method.entered can be used to ge this in single line

```
let _span_guard = span!(Level::INFO, "root", telemetry=true).entered();
```

Reply:
I'm going to be adding convenience macros for each event type, so it would also make sense to make one here too.

Reply:
A `span_telemetry!()` macro along with `<level>_telemetry!()` macros have been added.